modules/sdk-coin-stx/src/stx.ts

```diff
@@ -215,7 +215,7 @@ export class Stx extends BaseCoin {
           tokenName: findTokenNameByContract(txJson.payload.contractAddress, txJson.payload.contractName),
         };
         if (txJson.payload.functionArgs.length === 4) {
-          memo = txJson.payload.functionArgs[3].buffer.toString('ascii');
+          memo = Buffer.from(txJson.payload.functionArgs[3].value.buffer).toString();
```

Comment:
Can you please update the test for this as well, how was this not caught in tests